assignment-1/time_pipe.c

```diff
@@ -0,0 +1,59 @@
+#include <sys/types.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <iostream>
+#include <sys/wait.h>
+#include <sys/time.h>
+
+
+struct timeval_t
+{
+
+  time_t tv_sec;
+  suseconds_t tv_usec;
+
+timeval(int *elapsed_time)
+{
+  tv_sec = elapsed_time;
+  tv_usec = elapsed_time;
+
+}
+
+
+//-----------------I just put this function inside the struck//
+// I have to read teh chapther to understand it better. starting the program.
+// the I will have to
+// read the chapter to fully understand the better. .
+// its not finished yet. you will get an error if you cimpile.  thank you
+
+
+
+
+int main()
+{
+    pid = fork()
+
+    if (pid < 0)
+    {
+      fprintf(stderr, "Fork Failed");
+      return 1;
+    }
+    if (pid == 0)
+    {
+      timeval_t startTime;
+      gettimeofday( &startTime, 0 );
+    }

```

Comment:
This is a good start. Just have some minor changes to suggest from pg. 142 from the book.

```suggestion
/* create the pipe */
if (pipe(fd) == -1) {
    fprintf(stderr,"Pipe failed");
    return 1;
}

if (pid > 0) { /* parent process */
    /* close the unused end of the pipe */
    close(fd[READ_END]);
    
    /* write to the pipe */
    write(fd[WRITE_END], write_msg, strlen(write_msg)+1);
    
    /* close the write end of the pipe */
    close(fd[WRITE_END]);
} else { /* child process */
   /* close the unused end of the pipe */
   close(fd[WRITE_END]);
   
    /* read from the pipe */
    read(fd[READ_END], read msg, BUFFER_SIZE);
    
    printf("read %s",read_msg);
    /* close the read end of the pipe */
    
    close(fd[READ_END]);
}
```